packages/dd-trace/src/ci-visibility/exporters/agentless/writer.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+const request = require('./request')
+const log = require('../../../log')
+
+const { AgentlessCiVisibilityEncoder } = require('../../../encode/agentless-ci-visibility')
+
+class Writer {
```

Comment:
Similar to the above, can the original one be reused? Maybe with a base class?

Reply:
> makeRequest is quite different: it doesn't use 0.4/0.5 version input, trace count, lookup or needsStartupLog and doesn't set most of the headers the agent exporter does. We could extract the common pieces but I think the signature of the function would be unnecessarily complex for agentless writer.

Why is it that different? Other than being able to differentiate the version they should be very similar.

> _sendPayload operates with metrics and prioritySampler which we don't do in agentless.

Any reason not to? If everything is always sampled anyway it wouldn't make a difference to go through the sampler.

Reply:
> Why is it that different? Other than being able to differentiate the version they should be very similar.

they are _kind_ of similar but the agent one handles way more logic than ci visibility needs to. I'd rather not do a single method that includes multiple conditional logic. 

> Any reason not to? If everything is always sampled anyway it wouldn't make a difference to go through the sampler.

I think it makes sense to separate functions when the endpoint is different and returns different things, for example, the line where we read `res` and `rate_by_service` wouldn't make sense for us, so we'd have to handle conditionally (making the method awkward). The same can be said about `metrics`